src/ThreeWayMerge.php

```diff
@@ -6,52 +6,316 @@
         $merged = [];
-        $conflict = false;
         foreach ($ancestor as $key => $value) {
+            // Checks if the value contains an array itself.
             if (is_array($value)) {
                 $merged[$key] = $this->performMerge(
                     $value,
                     $local[$key],
                     $remote[$key]
                 );
             } else {
-                // If ancestor's value is equal to remote's value,
-                // Set the merged array's value to local value.
+                $merged[$key] = $this->merge(
+                    $ancestor[$key],
+                    $local[$key],
+                    $remote[$key],
+                    $key
+                );
+                //If a key doesn't have any value, unset the key.
+                if ($merged[$key] == null) {
+                    unset($merged[$key]);
+                }
+            }
+        }
+        return $merged;
+    }
 
-                // Else If ancestor's value is equal to local's value,
-                // Set the merged array's value to remote value.
+    /**
+     * Function to perform merge. Do not call it directly.
+     *
+     * @param string $x
+     * @param string $y
+     * @param string $z
+     * @param mixed $key
+     *
+     * @return mixed
+     * @throws Exception
+     */
+    protected function merge($x, $y, $z, $key)
+    {
+        // Convert the value into array.
+        $ancestor = (strpos($x, "\n") !== false ? explode("\n", $x) : array($x));
+        $local = (strpos($y, "\n") !== false ? explode("\n", $y) : array($y));
+        $remote = (strpos($z, "\n") !== false ? explode("\n", $z) : array($z));
 
-                // Else If local's value is equal to remote's value,
-                // Set the merged array's value to any value as they both are same.
+        // Count number of lines in value or elements in new formed array.
+        $count_ancestor = count($ancestor);
+        $count_remote = count($remote);
+        $count_local = count($local);
 
-                // Else set conflict to TRUE as none of the above is True.
-                if ($ancestor[$key] == $local[$key]) {
-                    $merged[$key] = $remote[$key];
-                } elseif ($ancestor[$key] == $remote[$key]) {
-                    $merged[$key] = $local[$key];
-                } elseif ($local[$key] == $remote[$key]) {
+        // If number of lines is different in all 3 values.
+        // For example : addition in remote and removal in local node.
+        //
+        // Else get the value of number of lines in updated node.
+        // For example : Addition of 2 lines in local.
+        // Suppose: $count_ancestor = 2 and $count_remote = 2
+        // then, $count_local would be 4 now.
+        if ($count_ancestor != $count_local
+            && $count_local!= $count_remote
+            && $count_ancestor != $count_remote
+        ) {
```

Comment:
I tried doing it but it seems like it Travis build won't pass. So I had to undo all the changes.